### ISSUE TITLE
[7.x] Prefer hot data tiers

### ISSUE THREAD

**Issue by jakelandis** [elastic/beats#30411]
Starting in [7.16](https://github.com/elastic/beats/pull/28538) Beats started using a composable index templates by default. Composable index templates will always take precedence over legacy index templates, meaning that an index patterns that match both composable and legacy templates, the legacy templates will be ignored. 

Normally this is fine and generally desired. However, prior to this change it was possible to declare a legacy index template to match on `*` (i.e. all indices) to default index settings to ensure that indices were first allocated on the hot node. Since Beats are now using composable index templates this type of pattern will no longer work. For systems which relied on that `*` pattern for hot allocation, they will now loose the ability to default to hot nodes. 

I would like to request the following index setting be the default for Beats 7.x :
```
"index.routing.allocation.include._tier_preference" : "data_hot,data_content"
```
This says prefer data_hot if present, else use data_content. If no node roles are configured for data tiers this is still OK and won't break anything since every node will logically get every data_* node role. However, if node roles are present then with this change new indices will prefer hot nodes. 

This is not necessary for 8.x defaults since 8.x Beats default to use data streams and data streams have an implicit default of data_hot.

**Comment by jlind23** [comment]
@nimarezainia @cmacknz do you think we should make it land in 7.17?

**Comment by cmacknz** [comment]
The change seems trivial to make, it just a question of having some available to do it.

Most of the effort would be in testing to confirm that the setting is applied as expected, and perhaps confirming that there are no backwards compatibility concerns as the description states.

**Comment by VimCommando** [comment]
This is still relevant, especially given 7.17 will be supported until 9.x is released; plus 7.17 is the recommended upgrade path to 8.x. This update helps smooth that transition.
